django/db/models/fields/related.py

```diff
@@ -134,6 +130,21 @@ def _check_related_name_is_valid(self):
                 )
             ]
         return []
+    
+    def _is_related_name_valid_id(self,related_name):
```

Comment:
Not sure a helper method is really needed. We could use a boolean like `related_name_invalid = False` and then have the conditions set it to `True`. and then replace `if related_name and not...` with `if related_name_invalid:`